### PR TITLE
Switch to 1ES R&D pools on main

### PR DESCRIPTION
**Pull request by jonfortescue**
Per https://github.com/dotnet/core-eng/issues/14276.

### CODE REVIEW

**Comment by jonfortescue** [comment]
The integration failures are because the variable name needs to be changed in the build variables.

**Comment by JoeRobich** [comment]
Updated Pool: NetCore1ESPool-Public
Updated Queue: Build.Windows.Amd64.VS2022.Pre.Open

**Comment by jonfortescue** [comment]
Here's hoping it just works.

**Comment by jonfortescue** [comment]
Looks like it's still looking for BuildPool for some reason...hmmm...

**Comment by JoeRobich** [comment]
integration corehost is a different pipeline. will update

**Comment by JoeRobich** [comment]
Didn't want to restart all the tests, so I kicked off a new corehost run for this PR - https://dev.azure.com/dnceng/public/_build/results?buildId=1454473&view=results

**Comment by JoeRobich** [comment]
Same for lsp integration - https://dev.azure.com/dnceng/public/_build/results?buildId=1454484&view=results

**Comment by JoeRobich** [comment]
Hmm a fix was merged into main for these integration failures. kicking off another run.

File: azure-pipelines-official.yml [hunk 1]
```diff
@@ -82,7 +82,7 @@ stages:
     displayName: Official Build
     timeoutInMinutes: 360
     pool:
-      name: NetCore1ESPool-Svc-Internal
+      name: NetCore1ESPool-Internal
       demands: ImageOverride -equals Build.Windows.Amd64.VS2022.Pre
```
Comment (JoeRobich) [inline azure-pipelines-official.yml:86]: Until certs are available.

```suggestion
      demands: ImageOverride -equals Build.Server.Amd64.VS2019
```

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**